Doc/library/typing.rst

```diff
@@ -1172,25 +1173,58 @@ These are not used in annotations. They are building blocks for creating generic
            """Return a list containing n references to x."""
            return [x]*n
 
-       def longest(x: A, y: A) -> A:
-           """Return the longest of two strings."""
-           return x if len(x) >= len(y) else y
 
-    The latter example's signature is essentially the overloading
-    of ``(str, str) -> str`` and ``(bytes, bytes) -> bytes``.  Also note
-    that if the arguments are instances of some subclass of :class:`str`,
-    the return type is still plain :class:`str`.
+       def print_capitalized(x: S) -> S:
+           """Print x capitalized, and return x"""
+           print(x.capitalize())
+           return x
+
+
+       def concatenate(x: A, y: A) -> A:
+           """Add two strings together."""
+           return x + y
+
+    Note that type variables can be *bound*, *constrained*, or neither, but
+    cannot be both bound *and* constrained.
+
+    Bound type variables and constrained type variables have different
+    semantics in several important ways. Using a *bound* type variable means
+    that the ``TypeVar`` will be solved using the most specific type possible::
+
+       x = print_capitalized('a string')
+       reveal_type(x)  # revealed type is str
+
+       class StringSubclass(str):
+           pass
+
+       y = print_capitalized(StringSubclass('another string'))
+       reveal_type(y)  # revealed type is StringSubclass
+
+       z = print_capitalized(45)  # error: int is not a subtype of str
+
+    Type variables can be bound to concrete types, abstract types (ABCs or
+    protocols), and even unions of types::
+
+       U = TypeVar('U', bound=str|bytes)  # Can be any subtype of the union str|bytes
```

Comment:
```suggestion
       U = TypeVar('U', bound=str | bytes)  # Can be any subtype of the union str | bytes
```
Conventional formatting

Reply:
Not sure about this one. The space around the `|` operator looks very odd to me, given the lack of space around the `=` operator. The semantic grouping is that the TypeVar is

`[bound to [str or bytes]]`

 not that the TypeVar is

`[[bound to str] or bytes]`

Reply:
But we also write `f(x=a + b)`. PEP 604 always puts spaces around `|`, though admittedly none of its examples are in a keyword argument.

Reply:
> But we also write `f(x=a + b)`.

Well, I personally would tend to avoid that as well :)

Reply:
(I'm happy to change the comment, btw, it's just the keyword arg I'm making a fuss about :)

Reply:
I thought about this some more and decided I don't care enough to argue about this any further :)

Reply:
I’m actually with Alex here. PEP 8 went too far IMO. :-)